### DISCUSSION TITLE
eval(..) 함수가 런타임 성능에 미치는 영향을 자세히 설명해 주세요.

### DISCUSSION THREAD

**Discussion by lledellebell** [diving-javascript/you-dont-know-js-yet#60] [1. 스코프] [answered]
<details>
<summary>관련 챕터</summary>
[Part2] 1.5 런타임에 스코프 변경하기
</details>

**Comment by wogus216** [comment, marked as answer]
<details>
<summary>답변</summary>


eval 함수는 크게 런타임 성능과 보안과 유지보수성에 영향을 끼칩니다.

**런타임 성능에 미치는 영향**

1. 컴파일 최적화 방해
2. 스코프와 변수  분석 방해
3. 코드 캐싱 방해

**보안과 유지보수성에 미치는 영향**

1. 보안문제
2. 디버깅과 유지보수의 어려움
3. 코드 가독성 저하
</details>
